crates/neon/src/types_impl/buffer/types.rs

```diff
@@ -0,0 +1,783 @@
+
+            // The borrowed data must be guarded by `Ledger` before returning
+            Ledger::try_borrow_mut(
+                &lock.ledger,
+                slice::from_raw_parts_mut(info.data.cast(), info.length),
+            )
+        }
+    }
+
+    fn size<'cx, C: Context<'cx>>(&self, cx: &mut C) -> usize {
+        self.len(cx) * std::mem::size_of::<Self::Item>()
+    }
+}
+
+impl<T: Binary> JsTypedArray<T> {
+    /// Constructs a typed array that views `buffer`.
+    ///
+    /// The resulting typed array has `(buffer.size() / size_of::<T>())` elements.
+    pub fn from_buffer<'cx, 'b: 'cx, C>(
+        cx: &mut C,
+        buffer: Handle<'b, JsArrayBuffer>,
+    ) -> JsResult<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        let size = buffer.size(cx);
+        let elt_size = std::mem::size_of::<T>();
+        let len = size / elt_size;
+
+        if (len * elt_size) != size {
+            panic!(
+                "byte length of typed array should be a multiple of {}",
+                elt_size
+            );
+        }
+
+        Self::from_region(cx, buffer.region(0, len))
+    }
+
+    /// Constructs a typed array for the specified buffer region.
+    ///
+    /// The resulting typed array has `region.len()` elements and a size of
+    /// `region.size()` bytes.
+    ///
+    /// Throws an exception if the region is invalid, for example if the starting
+    /// offset is not properly aligned, or the length goes beyond the end of the
+    /// buffer.
+    pub fn from_region<'c, 'r, C>(cx: &mut C, region: Region<'r, T>) -> JsResult<'c, Self>
+    where
+        C: Context<'c>,
+    {
+        let Region {
+            buffer,
+            offset,
+            len,
+            ..
+        } = region;
+
+        let result = unsafe {
+            sys::typedarray::new(cx.env().to_raw(), T::TYPE_TAG, buffer.to_raw(), offset, len)
+        };
+
+        if let Ok(arr) = result {
+            Ok(Handle::new_internal(Self(JsTypedArrayInner {
+                local: arr,
+                buffer: buffer.to_raw(),
+                _type: PhantomData,
+            })))
+        } else {
+            Err(Throw::new())
+        }
+    }
+
+    /// Returns information about the backing buffer region for this typed array.
+    pub fn region<'cx, C>(&self, cx: &mut C) -> Region<'cx, T>
+    where
+        C: Context<'cx>,
+    {
+        let env = cx.env();
+        let info = unsafe { sys::typedarray::info(env.to_raw(), self.to_raw()) };
+
+        Region {
+            buffer: Handle::new_internal(JsArrayBuffer::from_raw(cx.env(), info.buf)),
+            offset: info.offset,
+            len: info.length,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Constructs a new typed array of length `len`.
+    ///
+    /// The resulting typed array has a newly allocated storage buffer of
+    /// size `(len * size_of::<T>())` bytes.
+    pub fn new<'cx, C>(cx: &mut C, len: usize) -> JsResult<'cx, Self>
```

Comment:
I had one additional idea for a helper. `JsTypedArray::from_slice`.

```rust
impl<T: Binary> JsTypedArray<T> {
    pub fn from_slice<'cx, C>(cx: &mut C, data: &[T]) -> JsResult<'cx, Self>
    where
        C: Context<'cx>,
    {
        let buffer = cx.array_buffer(data.len() * std::mem::size_of::<T>())?;
        let mut array = Self::from_region(cx, buffer.region(0, data.len()))?;

        array.as_mut_slice(cx).copy_from_slice(data);

        Ok(array)
    }
}
```

```rust
let nums = [1, 2, 3, 4, 5];
let u32array = JsTypedArray::from_slice(&mut cx, nums);
```

Reply:
Agreed! I'll tackle that in a separate PR, as you suggest.